kratos/includes/properties.h

```diff
@@ -410,6 +740,43 @@ class Properties : public IndexedObject
     ///@name Private Operations
     ///@{
 
+    /**
+     * @brief This method trims a string in the different components to access recursively to any subproperty
+     * @param rStringName The given name to be trimmed
+     * @return The list of indexes
+     */
+    std::vector<IndexType> TrimComponentName(const std::string& rStringName)
```

Comment:
I don't think you need the non-const version

Reply:
OK